From f04874ea717f6a3116e5745e31c9126ac83fc5a1 Mon Sep 17 00:00:00 2001
From: Nicolas LOUBOUTIN <nicolas.louboutin@st.com>
Date: Mon, 29 Jul 2024 10:46:56 +0200
Subject: [PATCH] dts: use retram as permanent storage for debug

Use part of retram to enable ramoops trace (kernel and user logs).
Configure RIF in consequence (ca35 non-secure usage).


Signed-off-by: Nicolas LOUBOUTIN <nicolas.louboutin@st.com>
---
 .../arch/arm/dts/stm32mp257f-ev1-ca35tdcid-resmem.dtsi | 10 +++++++++-
 core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi   |  6 +++++-
 2 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-resmem.dtsi b/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-resmem.dtsi
index c5ec90130..f925fa402 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-resmem.dtsi
+++ b/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-resmem.dtsi
@@ -43,7 +43,15 @@
 		};
 
 		cm33_retram: cm33-retram@a080000 {
-			reg = <0x0 0xa080000 0x0 0x1f000>;
+			reg = <0x0 0xa080000 0x0 0x1b000>;
+			no-map;
+		};
+
+		ramoops_retram: ramoops@a09b000 {
+			compatible = "ramoops";
+			reg = <0x0 0xa09b000 0x0 0x4000>;
+			console-size = <0x1000>;
+			pmsg-size = <0x3000>;
 			no-map;
 		};
 
diff --git a/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi b/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi
index 793f077b4..d6e30fa20 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi
+++ b/core/arch/arm/dts/stm32mp257f-ev1-ca35tdcid-rif.dtsi
@@ -865,6 +865,10 @@
 	st,protreg = <RISABPROT(RIF_DDCID_DIS, RIF_UNUSED, RIF_SEC, RIF_NPRIV, RIF_CFDIS, RIF_UNUSED, RIF_UNUSED, RIF_UNUSED)>;
 };
 
+&ramoops_retram {
+	st,protreg = <RISABPROT(RIF_DDCID_DIS, RIF_UNUSED, RIF_NSEC, RIF_NPRIV, RIF_CFDIS, RIF_UNUSED, RIF_UNUSED, RIF_UNUSED)>;
+};
+
 &ddr_param {
 	st,protreg = <RISABPROT(RIF_DDCID_DIS, RIF_UNUSED, RIF_SEC, RIF_NPRIV, RIF_CFDIS, RIF_UNUSED, RIF_UNUSED, RIF_UNUSED)>;
 };
@@ -889,7 +893,7 @@
 
 &risab5 {
 	st,srwiad;
-	memory-region = <&cm33_retram>, <&ddr_param>;
+	memory-region = <&cm33_retram>, <&ramoops_retram>, <&ddr_param>;
 };
 
 &mm_ospi1 {
-- 
2.34.1

